Error on build opencv in ml module

- OpenCV version: 3.x
- Host OS: Windows XP SP3 x86
- Compiler VS 2010 & CMake: 3.4.3

### In which part of the OpenCV library you got the issue?
- ml

### Expected behaviour

### Actual behaviour
Error on build opencv from git

### Additional description
On build from command line:
msbuild /p:configuration=debug OpenCV.sln

received error:
"C:\Lib\opencv\build_x86_pre\OpenCV.sln" (целевой объект по умолчанию) (1) ->
"C:\Lib\opencv\build_x86_pre\ALL_BUILD.vcxproj.metaproj" (целевой объект по умо
лчанию) (2) ->
"C:\Lib\opencv\build_x86_pre\modules\aruco\opencv_aruco.vcxproj.metaproj" (целе
вой объект по умолчанию) (20) ->
"C:\Lib\opencv\build_x86_pre\modules\calib3d\opencv_calib3d.vcxproj.metaproj" (
целевой объект по умолчанию) (21) ->
"C:\Lib\opencv\build_x86_pre\modules\features2d\opencv_features2d.vcxproj.metap
roj" (целевой объект по умолчанию) (24) ->
"C:\Lib\opencv\build_x86_pre\modules\ml\opencv_ml.vcxproj.metaproj" (целевой об
ъект по умолчанию) (35) ->
"C:\Lib\opencv\build_x86_pre\modules\ml\opencv_ml.vcxproj" (целевой объект по у
молчанию) (36) ->
(Целевой объект ClCompile) ->
  ..\..\..\sources\opencv\modules\ml\src\svmsgd.cpp(156): error C2668: 'cv::sqr
t' : ambiguous call to overloaded function [C:\Lib\opencv\build_x86_pre\modules
\ml\opencv_ml.vcxproj]

    Предупреждений: 2633
    Ошибок: 1

### Code example to reproduce the issue / Steps to reproduce the issue
After replacement
`multiplier = static_cast<float>(sqrt(samples.total()) / normValue);`
in svmsgd.cpp(156) to
`multiplier = static_cast<float>(sqrt(static_cast<double>(samples.total())) / normValue);`
build was successful.

--- a/modules/ml/src/svmsgd.cpp
+++ b/modules/ml/src/svmsgd.cpp
@@ -153,7 +153,7 @@ void SVMSGDImpl::normalizeSamples(Mat &samples, Mat &average, float &multiplier)
 
     double normValue = norm(samples);
 
-    multiplier = static_cast<float>(sqrt(samples.total()) / normValue);
+    multiplier = static_cast<float>(sqrt(static_cast<double>(samples.total())) / normValue);
 
     samples *= multiplier;
 }

